library/src/main/java/com/mux/player/internal/cache/CacheDatastore.kt

```diff
@@ -393,31 +369,33 @@ internal class CacheDatastore(
   // If the db failed to open, this method will throw. Opening can be re-attempted after resolving
   @Throws(IOException::class)
   private fun awaitDbHelper(): DbHelper {
-    fun closeIfInterrupted(dbHelper: DbHelper?) {
+    // inner function closes db if the calling thread was interrupted
+    fun closeAndCancelIfInterrupted(dbHelper: DbHelper?) {
```

Comment:
Does this cancel the attempt to close the DB?

Reply:
It cleans up the database by closing the DBHelper in a timely fashion (since we potentially change its contents by evicting old entries). Then it cancels the task that was opening the database  and preparing the cache dir

On second thought, closing explicitly here might not really be necessary though. Like, by the same logic as above, that nothing will really be open